packages/connect-iframe/src/connectSettings.ts

```diff
@@ -31,40 +31,48 @@ export const parseConnectSettings = (
     // occasionally received origin is a stringified "null" or empty string. rename it to avoid confusion
     settings.origin = !origin || origin === 'null' ? 'unknown' : origin;
 
+    const hostOriginMatchesIframeOrigin = window.origin === settings.origin;
+
     // check if iframe is running on localhost
-    const isLocalhost = window?.location?.hostname === 'localhost';
+    const isIframeLocalhost = window?.location?.hostname === 'localhost';
     // check if origin is whitelisted
     const whitelist = isOriginWhitelisted(settings.origin);
 
-    settings.trustedHost = (isLocalhost || !!whitelist) && !settings.popup;
+    if (settings.trustedHost !== false) {
+        settings.trustedHost = isIframeLocalhost || !!whitelist;
+    }
 
     // ensure that popup will be used
     if (!settings.trustedHost) {
         settings.popup = true;
     }
 
     // ensure that debug is disabled
-    if (!settings.trustedHost && !whitelist && !isLocalhost) {
+    if (!settings.trustedHost && !isIframeLocalhost) {
         settings.debug = false;
     }
```

Comment:
this change, not sure if it is right. it is related to logging, so I am tagging @karliatto, we need to further figure out our logging strategy when we already have fancy logger now.

Reply:
oh, thats cool. let me just organize my thoughts - we already collect logs in sharedlogger regardless of `debug` setting right?

Reply:
Yes.